### ISSUE TITLE
html.elements.mark - Support in screen readers is mixed

### ISSUE THREAD

**Issue by aardrian** [mdn/browser-compat-data#21847]
### What type of issue is this?

Missing compatibility data

### What information was incorrect, unhelpful, or incomplete?

The chart considers only browser support, which can lead developers to believe the element is ok to use as-is.

However, support within screen readers is mixed.

### What browsers does this problem apply to, if applicable?

Chromium (Chrome, Edge 79+, Opera, Samsung Internet), Firefox, Safari

### What did you expect to see?

Some acknowledgement of mixed screen reader support.

### Did you test this? If so, how?

Yes. I used current releases of browsers and screen readers.

Test page:
https://codepen.io/aardrian/pen/rNRxZmP

Results of testing:

* JAWS/Chrome
    Precedes with “mark”, follows with “mark”.
* NVDA/Firefox
    Precedes with “highlighted”, follows with “out of highlighted”.
* Narrator/Edge
    Precedes with a pause, follows with a pause.
* Android/TalkBack/Chrome
    Precedes with a pause, follows with “highlight”.
* macOS/VoiceOver/Safari
    Precedes with “highlighted”, follows with a pause.
* iPadOS/VoiceOver/Safari
    Nothing exposed in read-all. Otherwise precedes with a pause, follows with “highlighted” 

### Can you link to any release notes, bugs, pull requests, or MDN pages related to this?

No, because I just tested it based on someone telling me the MDN chart said `<mark>` was good to use.

### Do you have anything more you want to share?

I know that screen reader support is out of scope for the support chart, but it should at least guide users that the chart only relates to visual rendering, not necessarily exposure to AAPIs and/or how that is then exposed to users.

### MDN URL

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/mark

### MDN metadata

<details>
<summary>MDN page report details</summary>

* Query: `html.elements.mark`
* Report started: 2024-01-04T23:30:34.911Z

</details>

**Comment by queengooborg** [comment]
Hi @aardrian!  At the moment, we are not planning to record compatibility data for screen readers in BCD, simply because we do not have the bandwidth to be able to do so at this time.  There is another project, however, that _does_ track this information, and we recommend checking it out: https://a11ysupport.io/ (there has been a bit of discussion about displaying this data on MDN, but the discussion has unfortunately not progressed)

**Comment by aardrian** [comment]
Fair enough.

I am familiar with a11ySupport.io (I have contributed). I felt with Google's Baseline effort it might consider 1) supporting more information here or 2) support a disclaimer here and support a11ySupport.io:
https://github.com/web-platform-dx/web-features/issues/498

So far no commitment has been made, so here I am filing issues.

**Comment by queengooborg** [comment]
It's a great topic to file issues for and get discussion going!  😄 

I'd say that the biggest issue right now is the bandwidth to add and maintain this data in BCD, or to maintain the data on a11ysupport.io.  While we have lots of amazing contributors helping keep the project maintained, there's only a handful of users with write access reviewing and merging PRs (currently 3-4, myself included), and we are all working to meet the goals for our existing projects (see the [Open Web Docs blog post](https://openwebdocs.org/content/posts/sovereign-tech-fund/) for more details).

Perhaps, however, you would be interested in [drafting a project proposal for Open Web Docs](https://github.com/openwebdocs/project/issues)?  This would get it into discussion during OWD's regular project planning, and perhaps a project we decide to tackle for the quarter/half year!

**Comment by aardrian** [comment]
So Open Web Docs has no overlap with the [WebDX Community Group](https://www.w3.org/community/webdx/), the ones driving Google's "Baseline" effort to [drive compat details into MDN](https://developer.mozilla.org/en-US/docs/Glossary/Baseline/Compatibility)?

Or, if it does have overlap, is OWD the place to raise that when WebDX is actively adding this badging to MDN?

Am I the only one confused here?

Because wow is this a lot of effort from me in my free time without the backing of Google, Apple, Microsoft, etc. to just try to understand who to talk to in order to make the materials here more accurate.

**Comment by queengooborg** [comment]
Sorry for the confusion.  Let me try to explain the relationships between the organizations and Baseline+BCD:

- [Open Web Docs](https://openwebdocs.org/) is an open source collective of technical writers.
  - OWD primarily contributes to MDN Web Docs projects, including MDN content and BCD.
  - OWD is governed by a Governing Committee and a Steering Committee.  The Steering Committee helps plan what projects OWD works on.
  - The 3-4 BCD maintainers with write access currently working on BCD are all from Open Web Docs, myself included.
      - Two of the owners of BCD are OWD team members, and a third owner is a part of OWD's steering committee.
- The [WebDX Community Group](https://www.w3.org/community/webdx/) is a group aimed to improve developer experience.
  - This community group does so primarily with their Baseline project.
  - A number of participants in this group are also members of the OWD Steering Committee, so they help plan the projects.
- Baseline is essentially a system to summarize compatibility information (particularly from BCD), which is displayed on MDN Web Docs.

I'm recommending creating an Open Web Docs project proposal because the main driving force working on BCD is Open Web Docs.  The project could be about adding the data into BCD, or it could even be about helping a11ysupport.io.  A project proposal will better help us understand the effort required for the project and let us track it as a goal to work on.

> when WebDX is actively adding this badging to MDN?

Are they?  From the [issue linked above](https://github.com/web-platform-dx/web-features/issues/498#ref-issue-2066499303), I haven't really seen any specific statement that Baseline will be adding this data; I've only seen mentions of how the data can be placed into BCD.  (I could be wrong though, it's a very long thread.)

---

I hope that this helps clarify things better.  I apologize if it didn't, I'm running on very little sleep and even less caffeine at the moment.

**Comment by aardrian** [comment]
Wow, that is _incredibly_ helpful. I appreciate you laying it all out. I still have to grok it (also low sleep, but also too much oatmeal).

I'll have to roll this around in my head and think about when/if I can file a proposal. Partly because it will have to be in my spare time. And I want to make sure if I do so, that is it not dismissed on some technicality of not knowing the process or whatever.

> Are they?

So far I only [saw it on `<dialog>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog), but I also am not actively looking.

**Comment by queengooborg** [comment]
> So far I only [saw it on `<dialog>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog), but I also am not actively looking.

Oh do you mean the ["Accessibility concerns"](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog#accessibility_concerns) section?  That was [added by Eric Bailey (GitHub)](https://github.com/mdn/content/pull/13383) and then updated by [Estelle Weyl (OWD) + Dipika (MDN)](https://github.com/mdn/content/pull/29195)!  While members of the WebDX group may contribute to MDN pages, the group itself is only responsible for the Baseline headers up at the top!

> I'll have to roll this around in my head and think about when/if I can file a proposal. Partly because it will have to be in my spare time. And I want to make sure if I do so, that is it not dismissed on some technicality of not knowing the process or whatever.

Certainly!  I can say that for the most part, we'll fix up any technicalities and such -- my own OWD project proposals are a great example of that.  😆

**Comment by aardrian** [comment]
> Oh do you mean the ["Accessibility concerns"](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog#accessibility_concerns) section?

Nope, I meant the big WebDX / Google Baseline badge at the top of the page. What you refer to as the header. Which you confirm is them, not OWD. Which is part of where my confusion comes from.

As an aside, I am aware of Eric's work on that page. And not just because he referenced it in the issue I linked way above.

**Comment by queengooborg** [comment]
Hmm, I'm not seeing any mention about accessibility in the Baseline header:

![image](https://github.com/mdn/browser-compat-data/assets/5179191/c2ce5523-c3d3-4bbd-85b1-8bf2b5448ea2)

**Comment by aardrian** [comment]
Exactly, which is why I filed https://github.com/web-platform-dx/web-features/issues/498 last month.

**Comment by tidoust** [comment]
> Nope, I meant the big WebDX / Google Baseline badge at the top of the page. What you refer to as the header. Which you confirm is them, not OWD.

For the sake of clarity, if you're looking for a clear owner for the header, it's not WebDX (or Google) either. The WebDX group is responsible for the Baseline definition, Baseline data (currently derived from BCD data) and Baseline logos. The group is happy to discuss what should appear in a Baseline banner, but the integration in MDN pages, and the actual design and location of the banner, is up to MDN. Same thing for the integration in Can I Use, which is up to Can I Use. The overlap of participants between the organizations makes boundaries fuzzy but it helps ensure that we remain aligned on goals and directions, and are aware of ongoing discussions on Baseline-related topics!

**Comment by queengooborg** [comment]
> For the sake of clarity, if you're looking for a clear owner for the header, it's not WebDX (or Google) either.

I've gotta be honest, I feel that this explanation confuses things further rather than clarifies them, especially for new contributors.  If WebDX is responsible for the definition and data of Baseline, then I'd say that the simplest way to describe ownership of Baseline is to say that the WebDX group is the owner of Baseline, and that MDN/Can I Use are responsible for how Baseline is rendered on their own websites (in the same way that Can I Use is responsible for rendering BCD over there).

> Exactly, which is why I filed [web-platform-dx/web-features#498](https://github.com/web-platform-dx/web-features/issues/498) last month.

Ahh.  Okay, my confusion is that I hadn't actually seen any comment there that indicates work is underway, aside from the most recent comment that mentions a PR that states that a11y is not a Baseline goal.

**Comment by tidoust** [comment]
> I've gotta be honest, I feel that this explanation confuses things further rather than clarifies them, especially for new contributors. If WebDX is responsible for the definition and data of Baseline, then I'd say that the simplest way to describe ownership of Baseline is to say that the WebDX group is the owner of Baseline, and that MDN/Can I Use are responsible for how Baseline is rendered on their own websites (in the same way that Can I Use is responsible for rendering BCD over there).

Ah, sorry if that did not come out as clearly as I thought it would. I was not trying to clarify ownership of Baseline (the WebDX group is the owner of Baseline), but rather to explain who's responsible for "how Baseline is rendered" (MDN/Can I Use), because I understand @aardrian might also want to change the banner itself to add a disclaimer about accessibility.

Regardless, https://github.com/web-platform-dx/web-features/issues/498 seems a good place to coordinate the discussion on the relationship between Baseline and assistive technology, and on possible ways to convey that to developers. There is no resolution for now because the WebDX group as a whole hasn't discussed the issue yet.

**Comment by aardrian** [comment]
@queengooborg 
> > Exactly, which is why I filed https://github.com/web-platform-dx/web-features/issues/498 last month.
>
> Ahh. Okay, my confusion is that I hadn't actually seen any comment there that indicates work is underway, aside from the most recent comment that mentions a PR that states that a11y is not a Baseline goal.

When I left that comment, no work was underway. So it is not surprising that when you looked there was no work underway.

@tidoust 
> Ah, sorry if that did not come out as clearly as I thought it would. I was not trying to clarify ownership of Baseline (the WebDX group is the owner of Baseline), but rather to explain who's responsible for "how Baseline is rendered" (MDN/Can I Use), because I understand @aardrian might also want to change the banner itself to add a disclaimer about accessibility.

It was not clear to me. That being said, I _wanted_ Baseline itself to include accessibility notes. When I discovered was out of scope, I asked for a disclaimer. Who owns the banner, frankly, is moot to me. The banner _and_ all the promotional and support material around Baseline need that disclaimer, IMO. The banner is only one part of it.

I'll follow up on the Baseline issue I originally filed.

Sorry for the noise, @queengooborg, I am still learning this Whack-a-Mole process.

**Comment by Elchi3** [comment]
Heya, @queengooborg and @aardrian, I wrote up a project proposal for Open Web Docs based on this discussion. https://github.com/openwebdocs/project/issues/193

Feel free to comment over there and let me know your thoughts and if I captured the problem space and potential solutions correctly. Can't promise we will take on this work soon, but I wanted to get it written up for future consideration for our collective :)
